facturacion_mexico/facturacion_fiscal/tests/test_nota_credito_descuento_relacion_01.py

```
@@ -478,6 +531,102 @@ def test_bloquea_si_no_es_borrador(self):
+		"""Con 'Enable Discount Accounting' = OFF la conversión procede normalmente."""
+		doc = _fake_si()
+		res = self._run(doc, discount_accounting=False)
+		self.assertTrue(res["ok"])
+		self.assertTrue(all(r.income_account == CUENTA_DESC for r in doc["items"]))
+
+
+class TestRevertirADevolucion(FrappeTestCase):
+	"""Reversión descuento → devolución (motivo 03): restaura EXACTO desde el origen vía
+	`sales_invoice_item`; bloquea sin modificar si alguna línea no mapea; solo en borrador."""
+
+	def _run(self, doc, origen):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_run`

(ANN202)